### ISSUE TITLE
':source Session.vim' leaves tab open from previous session

### ISSUE THREAD

**Issue by syntaxman** [neovim/neovim#8328]
- `nvim --version`:  zsh/2 1014 % nvim --version
NVIM v0.2.3-890-gbe67d926c

- Vim (version: ) behaves differently?
vim-8.0.0197:   Behaves the same

- Operating system/version:  
Debian 9 (stretch)

- Terminal name/version:
rxvt-unicode v9.22
tmux 2.3

- `$TERM`:
tmux-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
1. use :mksession to create one or two sessions - ensure that there are multiple open tabs
2. load a session using any appropriate method
3. :source the second session 
       (or reload the original session)
       (or remotely use nvr -c 'source session-name')
Number 3 can be performed multiple times with the issue occurring every time with extraneous tabs accumulating with each session load.

```

### Actual behaviour
With tabs open in current session, sourcing a new session containing tabs results in unfocused old tabs remaining unclosed.  If both sessions contain multiple tabs, they will accumulate as sessions are switched.

### Expected behaviour
The help file describes that sourcing a new session will result in all current buffers being closed, and it would seem that the tabs would close with them.  Instead those buffers associated with unfocused tabs remain unclosed.

**Comment by mhinz** [comment]
Well, Vim doesn't really have sessions. A session file is merely a collection of commands that try to recreate the state of tabs, windows, and buffers from when `:mksession` was used.

So, if you `:source`, you will always just add things, since you're not "switching sessions". There is no such thing.

> The help file describes that sourcing a new session will result in all current buffers being closed

Where?

**Comment by mhinz** [comment]
@syntaxman 

I think the documentation is outdated. Maybe it worked that way at some point in the past (mind that Vim is _very_ old already), but it's not how it behaves nowadays. Since we merely merge the runtime patches (that includes the docs) from upstream Vim, you should [create an issue for it there](https://github.com/vim/vim/issues). When they change it, we'll merge that change as well.

And yup, `:%bd` is most probably what you want. When you do `nvim foo bar`, you'll end up with two buffers. `foo` gets loaded and is the one you see. `bar` is in the buffer list (`:ls`), but not loaded yet. That will happen if you switch to `bar` for the first time. `:%bunload` frees a bit of memory taken by the buffers, but Vim will load them again if you cycle through the buffers.

Basically `:%bdelete` does the same as `:%bunload`, but also removes the buffers from the buffer list.

**Comment by chrisbra** [comment]
>>    The help file describes that sourcing a new session will result in all current buffers being closed

> Where?

Probably the same issue, that has just been reported to [Vim](https://github.com/vim/vim/issues/2864)
